src/status_im2/contexts/onboarding/sign_in/view.cljs

```diff
@@ -1,42 +1,223 @@
-   [quo/page-nav
-    {:align-mid?            true
-     :mid-section           {:type :text-only :main-text ""}
-     :left-section          {:type                :blur-bg
-                             :icon                :i/arrow-left
-                             :icon-override-theme :dark
-                             :on-press            #(rf/dispatch [:navigate-back])}
-     :right-section-buttons [{:type                :blur-bg
-                              :icon                :i/info
-                              :icon-override-theme :dark
-                              :on-press            #(js/alert "Pending")}]}]])
-
-(defn page
+(def camera (reagent/adapt-react-class CameraKitCamera))
+
+(defn- header
+  [active-tab read-qr-once?]
+  [rn/view
+   [rn/view {:style style/header-container}
+    [quo/button
+     {:icon                true
+      :type                :blur-bg
+      :size                32
+      :accessibility-label :close-sign-in-by-syncing
+      :override-theme      :dark
+      :on-press            #(rf/dispatch [:navigate-back])}
+     :i/close]
+    [quo/button
+     {:before              :i/info
+      :type                :blur-bg
+      :size                32
+      :accessibility-label :find-sync-code
+      :override-theme      :dark
+      :on-press            #(js/alert "Yet to be implemented")}
+     (i18n/label :t/find-sync-code)]]
+   [quo/text
+    {:size   :heading-1
+     :weight :semi-bold
+     :style  style/header-text}
+    (i18n/label :t/sign-in-by-syncing)]
+   [quo/text
+    {:size   :paragraph-1
+     :weight :regular
+     :style  style/header-sub-text}
+    (i18n/label :t/synchronise-your-data-across-your-devices)]
+   [rn/view {:style style/tabs-container}
+    [quo/segmented-control
+     {:size           32
+      :override-theme :dark
+      :blur?          true
+      :default-active @active-tab
+      :data           [{:id 1 :label (i18n/label :t/scan-sync-qr-code)}
+                       {:id 2 :label (i18n/label :t/enter-sync-code)}]
+      :on-change      (fn [id]
+                        (reset! active-tab id)
+                        (reset! read-qr-once? false))}]]])
+
+(defn- enable-camera-access-view
+  [request-camera-permission]
+  [rn/view {:style style/camera-permission-container}
+   [quo/text
+    {:size   :paragraph-1
+     :weight :medium
+     :style  style/enable-camera-access-header}
+    (i18n/label :t/enable-access-to-camera)]
+   [quo/text
+    {:size   :paragraph-2
+     :weight :regular
+     :style  style/enable-camera-access-sub-text}
+    (i18n/label :t/to-scan-a-qr-enable-your-camera)]
+   [quo/button
+    {:before              :i/camera
+     :type                :primary
+     :size                32
+     :accessibility-label :request-camera-permission
+     :override-theme      :dark
+     :on-press            request-camera-permission}
+    (i18n/label :t/enable-camera)]])
+
+(defn- calculate-qr-scan-hole-area
```

Comment:
The function name is a bit confusing. It says *calculate*, which makes it look like it'll return a calculation, but instead it returns hiccup. Do you have an idea of a name that represents what this component is (not what it does)?

Reply:
It doesn't calculate anything except it finds the coordinates (x,y) of itself in the screen, which is later used for holes in the `hole-view`.

I updated the name 👍. Apologies for the confusion.